src/gangliad/statsd.cpp

```diff
@@ -963,20 +976,19 @@ StatsD::publishMetrics( int /* timerID */ )
 	}
 
 	// Set query constraint to only fetch ad types needed
-	if( !m_target_types.contains_anycase("any") ) {
+	if (!contains_anycase(m_target_types,"any")) {
 		if( !m_requirements.empty() ) {
 			query.addANDConstraint(m_requirements.c_str());
 		}
 	}
 	else {
```

Comment:
Please pay particular attention to the follow block of code, as I believe there was a bug on the trunk -- a missing call to `rewind`.  The new code assumes there was a rewind there.

Reply:
Agreed that it looks like a `rewind()` is missing. And I think the `contains_anycase()` conditional is backwards. Check out commit 30663964e20.

Reply:
Huh.  Look at that.  We need to push for a good regression test for the gangliad, maybe even if it just writes ads it pulls out of the collector to a file.

Reply:
Almost feel like the condition should be something like `m_target_types.size() == 1 && m_target_types[0] == "any"`.

Seems odd to have "any" mixed in with other types.

Reply:
That seems like a mistake in the config. But I wouldn't ignore the 'any' because there are other types in the list.